R/mobr.R

```diff
@@ -10,221 +10,244 @@ require(pracma)
+    for (i in 1:nrow(plot_grp)) {
+        focal_site = plot_grp$plot[i]
         dist_to_site = pair_dist[i, ]
         # Shuffle plots, so that tied grouping is not biased by original order.
-        new_order = sample(1:nrow(plot_trmt))  
-        # new_order = 1:nrow(plot_trmt) # Alternative: no shuffles
-        plots_new = plot_trmt[new_order, 1]
+        new_order = sample(1:nrow(plot_grp))  
+        # new_order = 1:nrow(plot_grp) # Alternative: no shuffles
+        plots_new = plot_grp$plot[new_order]
         dist_new = dist_to_site[new_order]
         plots_new_ordered = plots_new[order(dist_new)]
         # Move focal site to the front
         plots_new_ordered = c(focal_site, 
                               plots_new_ordered[plots_new_ordered != focal_site])  
-        sp_ordered = sp_trmt[match(sp_trmt[ , 1], plots_new_ordered), 2:ncol(sp_trmt)]
+        sp_ordered = sp_grp[match(row.names(sp_grp), plots_new_ordered), ]
         # 1 for absence, 0 for presence
-        sp_bool = as.data.frame(ifelse(sp_ordered[ , 1:ncol(sp_ordered)] == 0, 1, 0)) 
+        sp_bool = as.data.frame((sp_ordered == 0) * 1) 
         rich = cumprod(sp_bool)
-        explic_loop[ , i] = as.numeric(ncol(dat_sp) - 1 - rowSums(rich))
+        explicit_loop[ , i] = as.numeric(ncol(dat_sp) - 1 - rowSums(rich))
     }
-    explic_S = apply(explic_loop, 1, mean)
-    return(explic_S)
+    explicit_S = apply(explicit_loop, 1, mean)
+    return(explicit_S[effort])
 }
 
-rarefy_sample_implicit = function(dat_sp, dat_plot, treatment) {
-    plot_trmt = dat_plot[dat_plot[ , 2] == treatment, ]
-    sp_trmt = dat_sp[match(plot_trmt[ , 1], dat_sp[ , 1]), ]
-    sample_S = rarefaction.sample(sp_trmt[ , 2:ncol(sp_trmt)])[ , 2]
+rarefy_sample_implicit = function(dat_sp, dat_plot, group, effort=NULL) {
+    sp_grp = dat_sp[dat_plot$group == group, ]
+    sample_S = rarefaction(sp_grp, 'samp', effort)
     return(sample_S)
 }
 
-rarefy_individual = function(dat_sp, dat_plot, treatment) {
-    # Returns the rarefaction results at each sample size
-    sp_trmt = dat_sp[match(dat_plot[dat_plot[ , 2] == treatment, 1], dat_sp[ , 1]), ]
-    trmt_sad = as.numeric(apply(sp_trmt[ , 2:ncol(sp_trmt)], 2, sum))
-    trmt_sad = trmt_sad[trmt_sad != 0]
-    ind_S = rarefaction.individual(trmt_sad)[ , 2]
+rarefy_individual = function(dat_sp, dat_plot, group, effort=NULL) {
+    sad = colSums(dat_sp[dat_plot$group == group, ])
+    ind_S = rarefaction(sad, 'indiv', effort)
     return(ind_S)
 }
 
-get_deltaSsad = function(dat_sp, dat_plot, treatment1, treatment2) {
-    rescaled_ind = sapply(c(treatment1, treatment2), function(x) 
-                          rarefy_individual(dat_sp, dat_plot, x))
-    Nind = min(length(unlist(rescaled_ind[1])), length(unlist(rescaled_ind[2])))
-    deltaSsad = as.numeric(unlist(rescaled_ind[2])[1:Nind] - 
-                           unlist(rescaled_ind[1])[1:Nind])
+get_deltaSsad = function(dat_sp, dat_plot, groups, Nind=NULL) {
```

Comment:
Agreed. Initially when I made the switch from `treatment1` and `treatment2` to `groups` I had hope of generalizing our code so that it could be run on any number of groups easily, but this is a good example of where that would break down at least with this set of code.